### PR TITLE
Chore: Switch to eslint-scope from escope

### PR DESCRIPTION
**Pull request by corbinu**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:
Switches to eslint-scope for 4.0.0 per #8266

**What changes did you make? (Give an overview)**
Remove escope and replaced with eslint-scope

**Is there anything you'd like reviewers to focus on?**
Nope is a drop in replacement no changes to eslint

### CODE REVIEW

File: tests/lib/eslint.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ const TEST_CODE = "var answer = 6 * 7;",
 //------------------------------------------------------------------------------
 
 /**
- * Get variables in the current escope
+ * Get variables in the current eslint-scope
```
Comment (not-an-aardvark) [inline tests/lib/eslint.js]: I think this should just say "the current scope".

**Comment by corbinu** [comment]
@not-an-aardvark It looks like the failures are only in phantomjs. Is it possible that our upgrading escope to run only on Node 4 included some JS that Phantom can't handle?

**Comment by not-an-aardvark** [comment]
You're right, it seems to be an issue with our build. I think it's because we're running babelify [here](https://github.com/eslint/eslint/blob/602e9c2a08c08f5827b34e075846b35770c15706/Makefile.js#L770) to get ESLint to work with PhantomJS, but it's not getting run on dependencies for some reason.

**Comment by not-an-aardvark** [comment]
Changing the `-t` in [this line](https://github.com/eslint/eslint/blob/602e9c2a08c08f5827b34e075846b35770c15706/Makefile.js#L770) to `--global-transform` seems to fix the build. I'm not sure if there's a better way to do this.

**Comment by corbinu** [comment]
Well I can make that change if that's what you think is best. I might also propose maybe we should switch out phantomjs for electron: https://github.com/twolfson/karma-electron/blob/master/README.md

**Comment by not-an-aardvark** [comment]
I'm not really familiar with the advantages/disadvantages of the tooling options here. For this PR, I think it would probably be better to keep phantomjs. Then we can switch to electron in another PR if necessary. (I want to keep the commits as atomic as possible.)

**Comment by alberto** [comment]
@corbinu We are using phantomjs to ensure eslint runs fine in a browser environment. I don't think `karma-electron` gives us that.

**Comment by mysticatea** [comment]
`electron` does not a headless browser, so it requires virtual display to run in Travis CI. I'm not sure the choice is proper.
I'd like vote to the use of global transform of `browserify`. (this implies we have to use the global transform for our online demo to support IE11)

**Comment by soda0289** [comment]
PhantomJS 2.5 beta supports es6 features like `const` if you want to give that a shot.
https://github.com/ariya/phantomjs/issues/14506
https://bitbucket.org/ariya/phantomjs/downloads/

**Comment by not-an-aardvark** [comment]
The issue isn't really that PhantomJS doesn't support ES6, it's that our bundle contains ES6. So things like the demo wouldn't work correctly on old browsers.

**Comment by soda0289** [comment]
Could we transform eslint-scope during publish? I think it used to use babel to do this but it was removed since we were only targeting node v4

**Comment by not-an-aardvark** [comment]
Yes, I think that's effectively what `--global-transform` does with browserify.

**Comment by corbinu** [comment]
@not-an-aardvark No was not proposing switching to electron in this commit just as a general solution. 

I guess I did not know eslint supported browsers at all. What browser versions do we need to support?

**Comment by not-an-aardvark** [comment]
I don't think we officially support anything in browsers, it's just that we need the [demo](http://eslint.org/demo) to work in ES5 browsers.

**Comment by corbinu** [comment]
Ahh ok so basically IE11 is the only issue (I feel like we all are going to spend the next 3 years saying that). Ya global transform is what you want then will push and we should be good

File: Makefile.js [hunk 2]
```diff
@@ -554,8 +554,8 @@ target.lint = function() {
 };
 
 target.test = function() {
-    target.lint();
-    target.checkRuleFiles();
+    //target.lint();
+    //target.checkRuleFiles();
```
Comment (not-an-aardvark) [inline Makefile.js]: Make sure to uncomment these :smile:

Comment (corbinu) [inline Makefile.js]: oops sorry was debugging thanks

**Comment by ilyavolodin** [comment]
If IE11 is giving us trouble, we could consider dropping support for it for demo (just need to add code for non-supported browsers). In the last 3 month, IE was only 1.08% of browsers visiting the site. And while that's still a pretty large number (around 5.8k sessions), demo has a lot less people visiting it. For demo page only IE is 1.1% of visits, but it's also only 105 pageviews in the last 3 month. So I think that could allow us to drop support for ES5 browsers (and remove babel from our test scripts).

**Comment by not-an-aardvark** [comment]
I agree that we shouldn't do additional work to support ES5 browsers. But in this case, if we can change a line of our build script to continue to support them without much extra development effort, it seems worthwhile.

**Comment by ilyavolodin** [comment]
@not-an-aardvark Agree. This was mostly about people trying to wire in babel into our site, so we could support ES6. If we want to do that, I'd rather drop IE support, then create a full blown build for the site. And if we were to do that, we might as well drop babel from our tests.

**Comment by not-an-aardvark** [comment]
Is this a breaking change? @ilyavolodin I noticed you labeled this as breaking, but I don't think it's breaking unless there's something I'm missing

**Comment by ilyavolodin** [comment]
@not-an-aardvark It's not breaking per-say (or at least we are hoping it's non-breaking), but we want to do this with 4.0 release, so I labeled it as `breaking` for consistency.

**Comment by platinumazure** [comment]
@not-an-aardvark In the original issue, I believe nzakas had said there was the theoretical possibility of breakage as the internals change, so he wanted it to be considered a breaking change just in case. (More or less; he phrased it differently.)

**Pull request merged**